scripts/src/scverse_template_scripts/cruft_prs.py

```diff
@@ -0,0 +1,171 @@
+  add `skip: true` to [`template-repos.yml`][] using a PR or,
+  if you never want to sync from the template again, delete your `.cruft` file.
+* If there are **merge conflicts**,
+  they either show up inline (`>>>>>>>`) or a `.rej` file will have been created for the respective files.
+  You need to address these conflicts manually. Make sure to enable pre-commit.ci (see below) to detect such files.
+* The scverse template works best when the [pre-commit.ci][], [readthedocs][] and [codecov][] services are enabled.
+  Make sure to activate those apps if you haven't already.
+
+[template-repos.yml]: https://github.com/scverse/ecosystem-packages/blob/main/template-repos.yml
+[pre-commit.ci]: {template_usage}#pre-commit-ci
+[readthedocs]: {template_usage}#documentation-on-readthedocs
+[codecov]: {template_usage}#coverage-tests-with-codecov
+"""
+
+
+class RepoInfo(TypedDict):
+    url: str
+    skip: NotRequired[bool]
+
+
+def get_template_release(gh: Github, tag_name: str) -> GHRelease:
+    template_repo = gh.get_repo("scverse/cookiecutter-scverse")
+    return template_repo.get_release(tag_name)
+
+
+def parse_repos(f: IO[str] | str) -> list[RepoInfo]:
+    repos = cast(list[RepoInfo], safe_load(f))
+    log.info(f"Found {len(repos)} known repos")
+    return repos
+
+
+def get_repo_urls(gh: Github) -> Generator[str]:
+    repo = gh.get_repo("scverse/ecosystem-packages")
+    file = cast(ContentFile, repo.get_contents("template-repos.yml"))
+    for repo in parse_repos(file.decoded_content):
+        if not repo.get("skip"):
+            yield repo["url"]
+
+
+def cruft_update(con: GitHubConnection, repo: GHRepo, path: Path, pr: PR) -> bool:
+    clone = Repo.clone_from(con.auth(repo.git_url), path)
+    branch = clone.create_head(pr.branch)
+    branch.checkout()
+
+    args = ["cruft", "update", " --checkout=main", "--skip-apply-ask", "--project-dir=."]
+    run(args, check=True)
+
+    if not clone.is_dirty():
+        return False
+
+    # Stage & Commit
+    # Maybe clean? changed_files = [diff.b_path or diff.a_path for diff in cast(list[Diff], clone.index.diff(None))]
+    # and then: clone.index.add([*clone.untracked_files, changed_files])
+    clone.git.add(all=True)
+    commit = clone.index.commit(pr.title, parent_commits=[branch.commit], author=con.sig, committer=con.sig)
+    branch.commit = commit
+
+    # Push
+    remote = clone.remote()
+    remote.set_url(con.auth(repo.clone_url))
+    remote.push([branch.name])
+    return True
+
+
+def make_pr(con: GitHubConnection, release: GHRelease, repo_url: str) -> None:
+    pr = PR(release)
+    log.info(f"Sending PR to {repo_url}: {pr.title}")
+
+    # create fork, populate branch, do PR from it
+    origin = con.gh.get_repo(repo_url.removeprefix("https://github.com/"))
+    repo = origin.create_fork()
+    with TemporaryDirectory() as td:
+        updated = cruft_update(con, repo, Path(td), pr)
+    if updated:
+        origin.create_pull(pr.title, pr.body, ...)  # TODO
```

Comment:
what happens if a pull request already exists from the last update but hasn't been merged yet?

Reply:
done